numpy/lib/tests/test_histograms.py

```diff
@@ -298,6 +298,20 @@ def test_datetime(self):
         assert_equal(d_edge.dtype, dates.dtype)
         assert_equal(t_edge.dtype, td)
 
+    def do_signed_overflow_bounds(self, dtype):
+        exponent = 8 * np.dtype(dtype).itemsize - 1
+        arr = np.array([-2**exponent + 4, 2**exponent - 4], dtype=dtype)
+        hist, e = histogram(arr, bins=2)
+        assert_equal(e, [-2**exponent + 4, 0, 2**exponent - 4])
+        assert_equal(hist, [1, 1])
+
+    def test_signed_overflow_bounds(self):
+        self.do_signed_overflow_bounds(np.byte)
+        self.do_signed_overflow_bounds(np.short)
+        self.do_signed_overflow_bounds(np.intc)
+        self.do_signed_overflow_bounds(np.int_)
+        self.do_signed_overflow_bounds(np.longlong)
```

Comment:
Poor-man's `pytest.parametrize`. Not written as a look because this way the stack trace is helpful

Reply:
So better than parametrize?

Reply:
Note that it is OK to use parametize.

Reply:
It matches the style that's already used in this file, so I'd be inclined to leave it as is, and then perhaps overhaul both cases in a later PR.